### DISCUSSION TITLE
Backpack website's currency format then switches

### DISCUSSION THREAD

**Discussion by Jenniedoll** [Laravel-Backpack/community-forum#989] [General]
The currency format of the [backpack](https://www.schoolbagsshop.com/) website is US dollars. How to add free conversion to other currencies?

**Comment by karandatwani92** [comment]
Hey @Jenniedoll 

This question is unrelated to BackpackForLaravel, but I run an e-commerce website, https://bulkmake.com, where I've built this currency conversion thing that gets currency updates for free.

Here is how I did it.
1) Used https://lyften.com/projects/laravel-currency/doc/ to display the amount in currency format. It also provides table migration. Unfortunately, its API for fetching currency rates is broken.
2) Created a Currency CRUD:
![Screenshot 2024-05-18 at 3 08 42 PM](https://github.com/Laravel-Backpack/community-forum/assets/8214221/eac0d554-a844-4e8c-850b-3931f05947c0)
2)  Created a scheduled command that updates the currency rate via the following API:
```
json_decode($this->request("https://api.apilayer.com/exchangerates_data/latest?base={$defaultCurrency}"));
```
3) Created a middleware to set display currency in session, selected via user or GeoIP.
